incubator/virtualcluster/pkg/controller/virtualcluster/virtualcluster_controller.go

```diff
@@ -185,28 +185,88 @@ func (r *ReconcileVirtualcluster) createPKI(vc *tenancyv1alpha1.Virtualcluster,
 	return nil
 }
 
+// genInitialClusterArgs generates the values for `--inital-cluster` option of etcd based on the number of
+// replicas specified in etcd StatefulSet
+func genInitialClusterArgs(replicas int32, stsName, svcName string) (argsVal string) {
+	for i := int32(0); i < replicas; i++ {
+		peerAddr := fmt.Sprintf("%s-%d=https://%s-%d.%s:2380", stsName, i, stsName, i, svcName)
+		if i == replicas-1 {
+			argsVal = argsVal + peerAddr
+			break
+		}
+		argsVal = argsVal + peerAddr + ","
+	}
+
+	return argsVal
+}
+
+// completmentETCDTemplate completments the ETCD template of the specified clusterversion
+// based on the virtual cluster setting
+func completmentETCDTemplate(vcName string, etcdBdl *tenancyv1alpha1.StatefulSetSvcBundle) {
+	etcdBdl.StatefulSet.ObjectMeta.Namespace = vcName
+	etcdBdl.Service.ObjectMeta.Namespace = vcName
+	args := &etcdBdl.StatefulSet.Spec.Template.Spec.Containers[0].Args
+	icaVal := genInitialClusterArgs(*etcdBdl.StatefulSet.Spec.Replicas,
+		etcdBdl.StatefulSet.Name, etcdBdl.Service.Name)
+	*args = append(*args, "--initial-cluster", icaVal)
+}
+
+// completmentETCDTemplate completments the apiserver template of the specified clusterversion
```

Comment:
s/completmentETCDTemplate/complementAPIServerTemplate/
please keep the comment sync with code😄